lib/daru/index.rb

```diff
@@ -96,6 +96,27 @@ def [](*key)
         v
       end
     end
+    
+    def pos *args
+      # Causes Segmentation fault in array_wrapper
+      if args.first.is_a? Range
+        return args.first.to_a
```

Comment:
I've tried to investigate it, and it seems like NMatrix garbage-collection problem (note this part of segfault info: https://gist.github.com/lokeshh/83369d71ca94b07cf89b23e215214666#file-errorslol-rb-L115, I've seen the same results on my investigation).

For moving further, I suggest to exclude NMatrix from ALL_TYPES for now, fix everything to work and then try again with NMatrix.